File: ANTIKEYLISTInputValidation.java
import java.util.Scanner;

public class ANTIKEYLISTInputValidation {

    // Define the approved characters for validation
    private static final String approvedChars = "3479ACDEFGHJKMNPQRTUVWXYacdefghjkmnpqrtuvwy@#$%^&*_+=";

    // Function to validate input based on the approved characters
    public static boolean validateInput(String userInput) {
        for (int i = 0; i < userInput.length(); i++) {
            if (approvedChars.indexOf(userInput.charAt(i)) == -1) {
                return false;  // Invalid character found
            }
        }
        return true;  // All characters are valid
    }

    // Main method to test input validation
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string to validate: ");
        String userInput = scanner.nextLine();

        // Validate user input
        if (validateInput(userInput)) {
            System.out.println("Input is valid.");
        } else {
            System.out.println("Input contains invalid characters.");
        }

        scanner.close();
    }
}

File: ANTIKEYLIST_input_validation.py
# Define the approved characters that are safe and not easily confused
approved_chars = "3479ACDEFGHJKMNPQRTUVWXYacdefghjkmnpqrtuvwy@#$%^&*_+="

# Function to validate input based on the approved character set
def validate_input(user_input):
    """
    Validates that a given input contains only approved characters.

    Parameters:
    user_input (str): The input string to validate

    Returns:
    bool: True if the input is valid (contains only approved characters), False otherwise
    """
    if not user_input:  # Check if input is empty
        return False

    return all(char in approved_chars for char in user_input)

# Main function for testing input validation
if __name__ == "__main__":
    user_input = input("Enter a string to validate: ")

    # Validate user input
    if validate_input(user_input):
        print("Input is valid.")
    else:
        print("Input contains invalid characters.")
